src/main/java/cn/edu/tsinghua/tsfile/encoding/decoder/Decoder.java

```diff
@@ -34,9 +34,13 @@ public static Decoder getDecoderByType(Encoding type, TSDataType dataType) {
             return new LongRleDecoder(EndianType.LITTLE_ENDIAN);
         } else if (type == Encoding.BITMAP && dataType == TSDataType.ENUMS) {
             return new BitmapDecoder(EndianType.LITTLE_ENDIAN);
-        } else if (dataType == TSDataType.FLOAT || dataType == TSDataType.DOUBLE) {
+        } else if ((dataType == TSDataType.FLOAT || dataType == TSDataType.DOUBLE) && (type == Encoding.RLE || type == Encoding.TS_2DIFF) ) {
             return new FloatDecoder(TSEncoding.valueOf(type.toString()), dataType);
-        }
+        } else if (type == Encoding.GORILLA && dataType == TSDataType.FLOAT) {
```

Comment:
Following code is not aligned, is it the problem of indentation or just displaying error of github？